File: lib/tailwind/create_tw_colors.js
const data = require("./color.json");

const colorToInt = (str) => {
  return str.replace("#", "0xff");
};

let core = "";

for (let key in data) {
  const primary = `_${key}PrimaryValue`;
  let name = key;
  if (key === "red" || key === "blue" || key === "green") {
    name += "Tw";
  }

  const str = `\n  static const int ${primary} = ${colorToInt(
    data[key][500]
  )};\n\n
  /// The slate primary color and swatch.
  ///
  /// ![](https://raw.githubusercontent.com/meetqy/flutter_tailwindcss/main/images/colors/${
    key.charAt(0).toUpperCase() + key.slice(1)
  }.png)
  ///
  static const MaterialColor ${name} =
        MaterialColor(_slatePrimaryValue, <int, Color>{
      50: Color(${colorToInt(data[key][50])}),
      100: Color(${colorToInt(data[key][100])}),
      200: Color(${colorToInt(data[key][200])}),
      300: Color(${colorToInt(data[key][300])}),
      400: Color(${colorToInt(data[key][400])}),
      500: Color(${primary}),
      600: Color(${colorToInt(data[key][600])}),
      700: Color(${colorToInt(data[key][700])}),
      800: Color(${colorToInt(data[key][800])}),
      900: Color(${colorToInt(data[key][900])}),
    });\n`;

  core += str;
}

const wrapper = `import 'package:flutter/material.dart';

class TwColor extends ColorSwatch<int> {
  const TwColor(int primary, Map<int, Color> swatch)
      : super(primary, swatch);

  ${core}
}
`;

require("fs").writeFileSync("../core/tw_colors.dart", wrapper);

File: lib/tailwind/create_tw_colors.dart
import 'dart:io' show File, Platform;
import 'package:path/path.dart' show dirname, join;
import 'dart:convert' show json;

String hexToColor(String code) {
  return code.replaceAll('#', '0xff');
}

void main() {
  String dir = join(dirname(Platform.script.path), 'colors.json');
  String saveDir =
      join(dirname(Platform.script.path), '../core/tw_colors.dart');
  Map data = json.decode(File(dir).readAsStringSync());

  String core = '';
  data.forEach((key, value) {
    String primary = '_${key}PrimaryValue';
    String name = key.toString();
    if (key == "red" || key == "blue" || key == "green") {
      name += "Tw";
    }

    String color = hexToColor(value['500'].toString());

    String str = '\nstatic const int $primary = $color;\n';
    str +=
        '/// The slate primary color and swatch.\n///\n/// ![](https://raw.githubusercontent.com/meetqy/flutter_tailwindcss/main/images/colors/$key.png);\n///\n';

    str +=
        'static const MaterialColor $name =MaterialColor(_slatePrimaryValue, <int, Color>{\n';
    str += '50: Color(${hexToColor(value["50"].toString())}),\n';
    str += '100: Color(${hexToColor(value["100"].toString())}),\n';
    str += '200: Color(${hexToColor(value["200"].toString())}),\n';
    str += '300: Color(${hexToColor(value["300"].toString())}),\n';
    str += '400: Color(${hexToColor(value["400"].toString())}),\n';
    str += '500: Color($primary),\n';
    str += '600: Color(${hexToColor(value["600"].toString())}),\n';
    str += '700: Color(${hexToColor(value["700"].toString())}),\n';
    str += '800: Color(${hexToColor(value["800"].toString())}),\n';
    str += '900: Color(${hexToColor(value["900"].toString())}),\n});\n';
    core += str;
  });

  String result =
      'import "package:flutter/material.dart";\nclass TwColor extends ColorSwatch<int> {\nconst TwColor(int primary, Map<int, Color> swatch)\n: super(primary, swatch);\n$core\n}';

  File(saveDir).writeAsStringSync(result);
}
